x-pack/plugin/security/qa/basic-enable-security/build.gradle

```diff
@@ -25,11 +24,35 @@ testClusters {
     numberOfNodes = 2
     setting 'xpack.ml.enabled', 'false'
     setting 'xpack.license.self_generated.type', 'basic'
-    setting 'xpack.security.enabled', 'false'
   }
 }
 
-tasks.register("javaRestTestWithSecurity", StandaloneRestIntegTestTask) {
+tasks.register("javaRestTestWithSecurityDisabled", StandaloneRestIntegTestTask) {
+  description = "Run tests against a cluster that has security explicitly disabled"
+  dependsOn "javaRestTest"
+  useCluster testClusters.javaRestTest
+  systemProperty 'tests.has_security', 'false'
+  testClassesDirs = sourceSets.javaRestTest.output.classesDirs
+  classpath = sourceSets.javaRestTest.runtimeClasspath
+  onlyIf {
+    BuildParams.inFipsJvm == false
+  }
+  doFirst {
+    testClusters.javaRestTest {
+      // TODO Rene: revisit if using dedicated new cluster definitions would be more efficient.
+      // Reconfigure cluster to explicitly disable security
+      setting 'xpack.ml.enabled', 'false'
+      setting 'xpack.license.self_generated.type', 'basic'
+      setting 'xpack.security.enabled', 'false'
+
+      restart()
+    }
+    nonInputProperties.systemProperty 'tests.rest.cluster', "${-> testClusters.javaRestTest.getAllHttpSocketURI().join(",")}"
+  }
+}
+tasks.named("check").configure { dependsOn("javaRestTestWithSecurityDisabled") }
+
+tasks.register("javaRestTestWithSecurityEnabled", StandaloneRestIntegTestTask) {
   description = "Run tests against a cluster that has security enabled"
   dependsOn "javaRestTest"
```

Comment:
In `EnableSecurityOnBasicLicenseIT` you rely on this task to run last.
I think you need:
```
mustRunAfter("javaRestTestWithSecurityDisabled")
```

Reply:
> Is the scenario this is supposed to cover, not covered elsewhere?
It is testing the actual change this PR is introducing - verifying the warning headers are there when security is implicitly disabled and that they are not present if security is explicitly enabled or disabled.